trl/trainer/grpo_trainer.py

```diff
@@ -845,23 +901,60 @@ def _generate_and_score_completions(
                     reward_inputs = super()._prepare_inputs(reward_inputs)
                     with torch.inference_mode():
                         rewards_per_func[:, i] = reward_func(**reward_inputs).logits[:, 0]  # Shape (B*G,)
+                    # For model-based rewards, we assume all samples are valid
+                    reward_masks[:, i] = True
                 else:
                     # Repeat all input columns (but "prompt" and "completion") to match the number of generations
                     keys = [key for key in inputs[0] if key not in ["prompt", "completion"]]
                     reward_kwargs = {key: [example[key] for example in inputs] for key in keys}
                     output_reward_func = reward_func(prompts=prompts, completions=completions, **reward_kwargs)
-                    rewards_per_func[:, i] = torch.tensor(output_reward_func, dtype=torch.float32, device=device)
```

Comment:
instead of relying on a mask, maybe a simpler approach is to convert None to nan:

```
output_reward_func = [x if x is not None else torch.nan for x in output_reward_func]
```